SDK/bundles/com.ashling.riscfree.debug.opxd.registers/src/com/ashling/riscfree/debug/opxd/registers/core/RiscFreeRegister.java

```diff
@@ -0,0 +1,749 @@
+							rm.setData(new IRegisterGroupDMData() {
+
+								@Override
+								public String getName() {
+									return "All Registers"; //$NON-NLS-1$
+								}
+
+								@Override
+								public String getDescription() {
+									return ""; //$NON-NLS-1$
+								}
+							});
+						} else {
+							rm.setData(dmData);
+						}
+						rm.done();
+					}
+				});
+	}
+
+	@Override
+	public void getRegisterData(IRegisterDMContext regDmc, DataRequestMonitor<IRegisterDMData> rm) {
+		super.getRegisterData(regDmc, new DataRequestMonitor<IRegisters.IRegisterDMData>(getExecutor(), rm) {
+			@Override
+			protected void handleSuccess() {
+				// This is to support register description from the register file
+				IRegisterDMData registerDMData = getData();
+				final IFrameDMContext frameDmc = DMContexts.getAncestorOfType(regDmc, IFrameDMContext.class);
+				if (registerDMData instanceof RegisterData && null != frameDmc) {
+					RegisterData regData = ((RegisterData) registerDMData);
+					String registerName = regData.getName();
+					String registerDescription = registerDescCache.getRegisterDescription(registerName);
+					if (null != registerDescription && !registerDescription.isEmpty()) {
+						rm.setData(new RegisterData(frameDmc, registerName, registerDescription, regData.isFloat()));
+					} else {
+						rm.setData(registerDMData);
+					}
+				} else {
+					rm.setData(new RegisterData(null, registerDMData.getName(), BLANK_STRING, false));
+				}
+				rm.done();
+			}
+		});
+	}
+
+	private long convertToLong(String formatId, String bitFieldValue) {
+		String formattedBitFieldVAlue = bitFieldValue;
+		if (formatId.equalsIgnoreCase(IFormattedValues.HEX_FORMAT)
+				|| formatId.equalsIgnoreCase(IFormattedValues.STRING_FORMAT)) {
+			if (!bitFieldValue.startsWith("0x")) { //$NON-NLS-1$
+				formattedBitFieldVAlue = "0x" + bitFieldValue; //$NON-NLS-1$
+			}
+		} else if (formatId.equalsIgnoreCase(IFormattedValues.OCTAL_FORMAT)) {
+			if (!bitFieldValue.startsWith("0")) { //$NON-NLS-1$
+				formattedBitFieldVAlue = "0" + bitFieldValue; //$NON-NLS-1$
+			}
+		}
+		long bitFieldVal;
+		if (formatId.equalsIgnoreCase(IFormattedValues.BINARY_FORMAT)) {
+			bitFieldVal = Long.parseLong(bitFieldValue, 2);
+		} else {
+			bitFieldVal = Long.decode(formattedBitFieldVAlue);
+		}
+		return bitFieldVal;
+	}
+
+	@Override
+	public void restoreDefaultGroups(IDMContext selectionContext, RequestMonitor rm) {
+		if (registerDescCache != null) {
+			registerGroups.clear();
+			registerGroups.addAll(registerDescCache.getRegisterGroups());
+		}
+		super.restoreDefaultGroups(selectionContext, rm);
+	}
+
+	private ILaunchConfiguration getLaunchConfig() {
+		ILaunch launch = (ILaunch) getSession().getModelAdapter(ILaunch.class);
+		if (launch == null) {
+			// The launch is no longer active
+			return null;
+		}
+		ILaunchConfiguration config = launch.getLaunchConfiguration();
+		return config;
+	}
+
+	public String getRegisterFile() {
```

Comment:
Need better function name since it's not just getting register file. Need proper comment, since we are doing some complex logic here.

Reply:
comments are added